main.java

```diff
@@ -0,0 +1,80 @@
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Scanner;
+
+public class main {
+    // Swapping two elements function
+    public static void swap(String[] array, int c, int j) {
+        String temp = array[c];
+        array[c] = array[j];
+        array[j] = temp;
+    }
+
+    // Initiate last elment as pivot, insert the pivot elment in the array in a sorted manner
+    public static int partition(String[] arr, int left, int right) {
+        // Setting pivot
+        String pivot = arr[right];
+
+        // Initiate the index of smaller element at rightmost position of pivot
+        int i = (left - 1);
+
+        for (int j = left; j <= right - 1; j++) {
+
+            // If current string is smaller than the pivot
+
+            if (arr[j].compareTo(pivot) < 0) {
```

Comment:
why use `compareTo` insetead of `if(arr[j] <= pivot)`

Reply:
`compareTo` is Java's built-in function inside of `String class` for performing strings comparison. Java doesn't recognize `arr[j] <= pivot)` as string operations and hence will show `bad operands` error.